sdk/core/core/inc/az_result.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_RESULT_H
+#define AZ_RESULT_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+  AZ_OK            =          0,
+  AZ_ERROR_FLAG    = 0x80000000,
+  AZ_STREAM_RESULT =    0x10000,
+  AZ_JSON_RESULT   =    0x20000,
+};
+
+typedef int32_t az_result;
```

Comment:
I presume this is the error type we discussed the other week.  I suggest adding comments that explain the layout of the bits and what they mean (doesn't have to block this PR).